app/controllers/api/v1/menu_controller.rb

```diff
@@ -0,0 +1,57 @@
+class API::V1::MenuController < ApplicationController
+
+  def index
+    # TODO: Unwrap child menu items into resulting json data
+    render :json => MenuItem.where(menu_item_id: nil), :include => [:menu_items]
+
+  end
+
+  def create
+    restrict_access
+
+    menu_item = MenuItem.new(item_params)
+    menu_item.save
+
+    redirect_to api_v1_menu_url(menu_item)
+  end
+
+  def show
+    render :json => MenuItem.find(params[:id])
+  end
+
+  def update
+    restrict_access
+
+    menu_item = MenuItem.find(params[:id])
+
+    if menu_item.update(item_params)
+      redirect_to api_v1_menu_url(menu_item)
+    else
+      raise 'Unable to save menu item'
+    end
+  end
+
+  def destroy
+    restrict_access
+
+    menu_item = MenuItem.find(params[:id])
+    menu_item.destroy
+
+    # What is preferably returned here?
+    render :json => []
```

Comment:
Consider resolving comment by returning an empty 204 response (see http://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete).